flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/dataformat/DataFormatConverters.java

```diff
@@ -128,8 +135,21 @@
 	 *                   lost its specific Java format. Only DataType retains all its
 	 *                   Java format information.
 	 */
-	@SuppressWarnings("unchecked")
+	@Deprecated
 	public static DataFormatConverter getConverterForDataType(DataType originDataType) {
+		return getConverterForDataType(originDataType, new Context(DEFAULT_TIME_ZONE));
```

Comment:
Shouldn't we use `TableConfig.getTimeZone` as the conversion time zone instead of local time zone in flink framework?

Reply:
After offline discussion with @wuchong , `java.sql.Timestamp` is a special format, `valueOf` and `toString` in `sql.Timestamp` will read the `TimeZone.getDefault`, So there, we should just use `TimeZone.getDefault` to correct `valueOf` and `toString`.